compare-evidence-strings/compare.sh

```diff
@@ -0,0 +1,128 @@
+wget -q https://github.com/theZiz/aha/archive/0.5.zip
+unzip -q 0.5.zip && cd aha-0.5 && make && mv aha ../ && cd .. && rm -rf aha-0.5 0.5.zip
+
+echo "Sort keys and remove schema version from the evidence strings"
+sort_keys "${OLD_EVIDENCE_STRINGS}" 01.keys-sorted.old.json \
+  & sort_keys "${NEW_EVIDENCE_STRINGS}" 01.keys-sorted.new.json \
+  & wait
+
+echo "Extract some fields to pair old and new evidence strings together"
+extract_fields 01.keys-sorted.old.json 02.fields.old \
+  & extract_fields 01.keys-sorted.new.json 02.fields.new \
+  & wait
+
+echo "Paste fields & original strings into the same table and sort"
+paste 02.fields.old 01.keys-sorted.old.json | sort -u > 03.fields-and-strings.old \
+  & paste 02.fields.new 01.keys-sorted.new.json | sort -u > 03.fields-and-strings.new \
+  & wait
+
+# Comparing functional consequences *correctly* will require investigating the nature of relationships (one-to-one,
+# one-to-many etc.) for all ClinVar objects. This will be done in subsequent tickets. Until then, functional consequence
+# calculations are disabled so as not to be misleading.
+
+#echo "Extract all unique variants"
+#cut -f1-2 03.fields-and-strings.old | sort -u > 04.all-variants.old \
+#  & cut -f1-2 03.fields-and-strings.new | sort -u > 04.all-variants.new \
+#  & wait
+#
+#echo "Group variants into deleted, new and common"
+#comm -23 04.all-variants.old 04.all-variants.new > 05.variants.deleted \
+#  & comm -13 04.all-variants.old 04.all-variants.new > 05.variants.added \
+#  & comm -12 04.all-variants.old 04.all-variants.new > 05.variants.common \
+#  & wait
+#
+#echo "Find variants which appear in both files but where functional mappings have changed"
+#cut -f1-3 03.fields-and-strings.old | sort -u > 06.consequences.old \
+#  & cut -f1-3 03.fields-and-strings.new | sort -u > 06.consequences.new \
+#  & wait
+#join 3_common old.consequences -j 1 | join /dev/stdin new.consequences -j1 | awk '$2 != $3' > 3_common_changed
+#
+#There are $(wc -l 1_deleted) records present only in the first file:
+#$(sed 's|^|  |' 1_deleted)
+#
+#There are $(wc -l 2_added) records present only in the second file:
+#$(sed 's|^|  |' 2_added)
+#
+#There are also $(wc -l 3_common) records present in both files.
+#Of them, $(wc -l 3_common_changed) records have different consequences between the first and the second files:
+#$(sed 's|^|  |' 3_common_changed)
+
+echo "Compute difference"
+git diff \
+  --minimal \
+  -U0 \
+  --color=always \
+  --word-diff=color \
+  03.fields-and-strings.old \
+  03.fields-and-strings.new &> "diff"
+
+echo "Producing the report"
+cat << EOF > report
+Compared:
+
+File 1
+${OLD_EVIDENCE_STRINGS}
+Total unique records: $(wc -l <03.fields-and-strings.old)
+
+File 2
+${NEW_EVIDENCE_STRINGS}
+Total unique records: $(wc -l <03.fields-and-strings.new)
+
+The full diff between two files follows.
+
+$(tail -n+5 diff | awk '{if ($0 !~ /@@/) {print $0 "\n"}}')
+EOF
+
+./aha --word-wrap < report > report.html
```

Comment:
Can this tool provide side-by-side diffs?

Reply:
`aha` just converts whatever input you had into HTML. The actual diff is produced by git diff, which cannot do side-by-side. However, for evidence strings, the word-diff mode with colouring might arguably even better than side-by-side.

When I next compare the evidence strings, I'll try `diff2html-cli`'s side-by-side view and see which one is better in the end.